accessible/variable-add-modal.component.js

```diff
@@ -0,0 +1,162 @@
+/**
+ * AccessibleBlockly
+ *
+ * Copyright 2017 Google Inc.
+ * https://developers.google.com/blockly/
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Component representing the variable rename modal.
+ *
+ * @author corydiers@google.com (Cory Diers)
+ */
+
+blocklyApp.VariableAddModalComponent = ng.core.Component({
+  selector: 'blockly-add-variable-modal',
+  template: `
+    <div *ngIf="modalIsVisible"class="blocklyModalCurtain"
+         (click)="dismissModal()">
+      <!-- $event.stopPropagation() prevents the modal from closing when its
+      interior is clicked. -->
+      <div id="varModal" class="blocklyModal" role="alertdialog"
+           (click)="$event.stopPropagation()" tabindex="0"
+           aria-labelledby="variableModalHeading">
+          <form id="varForm">
+            <p id="inputLabel">New Variable Name:
+              <input id="mainFieldId" type="text" [ngModel]="VALUE"
+                     (ngModelChange)="setTextValue($event)" tabindex="0"
+                     aria-labelledby="inputLabel" />
+            </p>
+            <hr>
+            <button id="submitButton" (click)="submit()">
+              SUBMIT
+            </button>
+            <button id="cancelButton" (click)="dismissModal()">
+              CANCEL
+            </button>
+          </form>
+      </div>
+    </div>
+  `,
+  pipes: [blocklyApp.TranslatePipe]
+})
+.Class({
+  constructor: [
+    blocklyApp.AudioService, blocklyApp.KeyboardInputService, blocklyApp.VariableModalService,
+    function(audioService, keyboardService, variableService) {
+      this.workspace = blocklyApp.workspace;
+      this.variableModalService = variableService;
+      this.audioService = audioService;
+      this.keyboardInputService = keyboardService
+      this.modalIsVisible = false;
+      this.activeButtonIndex = -1;
+
+      var that = this;
+      this.variableModalService.registerPreAddShowHook(
+        function() {

```

Comment:
Yes - I finally figured out yesterday how to move this all into a central place, which I'll be doing for my next commit.